sycl/include/CL/__spirv/spirv_types.hpp

```diff
@@ -129,10 +136,12 @@ enum class MatrixLayout : uint32_t {
 // information to SPIRV translator.
 // The long term solution would be to introduce a matrix type in Clang and use
 // it instead of this member.
-template <typename T, std::size_t R, std::size_t C, MatrixLayout U,
+template <typename T, std::size_t R, std::size_t C, MatrixUse U, MatrixLayout L,
```

Comment:
This matches the template arguments used for `joint_matrix` in matrix-tensorcore.hpp (the current CUDA impl), except that the ordering is a little different (MatrixUse is the second template param in matrix-tensorcore.hpp).  Trivial point, but we should align an ordering before merging the impls into a single `SYCL_EXT_ONEAPI_MATRIX` version.